tmt/steps/provision/mrack.py

```diff
@@ -104,84 +133,55 @@ def import_and_load_mrack_deps(workdir: Any, name: str) -> None:
     # error: Class cannot subclass "BeakerTransformer" (has type "Any")
     # as mypy does not have type information for the BeakerTransformer class
     class TmtBeakerTransformer(BeakerTransformer):  # type: ignore[misc]
-        def _parse_amount(self, in_string: str) -> Tuple[str, int]:
-            """ Return amount from given string """
-            result = []
-            amount = ""
-
-            for op in operators:
-                parts = in_string.split(op, maxsplit=1)
-                if len(parts) != 2:
-                    continue
-
-                if parts[0]:
-                    continue
+        def _translate_tmt_hw(self, hw: tmt.hardware.Hardware) -> Dict[str, Any]:
+            """ Return hw requirements from given hw dictionary """
 
-                if any(op in parts[1] for op in op):
-                    continue
+            assert hw.constraint
 
-                result.append(op)
-                amount = parts[1]
-                break
+            system = {}
+            disks = []
+            cpu = {}
 
-            assert len(result) == 1
-            assert len(amount) >= 1
+            for constraint in hw.constraint.variant():
+                if not isinstance(constraint, tmt.hardware.Constraint):
+                    raise tmt.utils.ProvisionError(f"Unsupported hw constraint '{constraint}'")
```

Comment:
```suggestion
                    raise tmt.utils.ProvisionError(f"Unsupported hardware constraint '{constraint}'.")
```

Reply:
Fixed in dc5b7deb394fcbec89dd5bc4197ad4c566022076